src/components/select/select.js

```diff
@@ -1,30 +1,65 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import PropTypes from "prop-types";
+import deprecated from "deprecated-prop-type";
+import warning from "warning";
 
 /**
  *
  * @param {string} label
  * @param {string} description
- * @param {[{ text: string, value: string, selected?:boolean }]}
- * options List of select options with key, value and selected properties
+ * @param {[{ text: string, value: string | number, disabled?: boolean }]} options
+ * List of select options with key, value and disabled properties
  * @param {object} selectProps Contains HTML input attributes:
  * type, value, name, id, etc. https://www.w3schools.com/tags/tag_select.asp
+ * @param { string | number } defaultValue
  */
-export const Select = ({ label, description, options, selectProps }) => {
+export const Select = ({
+  label,
+  description,
+  options,
+  selectProps,
+  defaultValue
+}) => {
+  const [currentValue, setCurrentValue] = useState();
+
+  useEffect(() => {
+    const selectedOptions = options.filter(opt => opt.selected);
+
+    if (selectedOptions.length) {
+      const [option] = selectedOptions;
+      setCurrentValue(option.value);
+
+      if (selectedOptions.length > 1) {
+        warning(false, "Only one 'selected' property of 'Select' can be true");
+      }
+    }
```

Comment:
If it's desired to have only one option selected, couldn't we just use `.find` instead of `.filter` ?

```js
    const selectedOptions = options.find(opt => opt.selected);
```

Reply:
Yes, but we won't know if we have more than one afterwards.